include/toml/toml.h

```diff
@@ -289,16 +289,24 @@ inline ParseResult parse(std::istream& is)
     return ParseResult(std::move(v), std::move(parser.errorReason()));
 }
 
-[[noreturn]]
-inline void failwith(const char* reason, ...)
+inline std::string format(std::stringstream &ss)
+{
+    return ss.str();
+}
+
+template<typename T, typename... Args>
+std::string format(std::stringstream &ss, T &&t, Args&&... args)
 {
-    char buf[1024];
-    va_list va;
-    va_start(va, reason);
-    vsnprintf(buf, 1024, reason, va);
-    va_end(va);
+    ss << std::forward<T>(t);
+    return format(ss, std::forward<Args>(args)...);
+}
 
-    throw std::runtime_error(buf);
+template<typename... Args>
+[[noreturn]]
+void failwith(Args&&... args)
+{
+    std::stringstream ss;
+    throw std::runtime_error(format(ss, std::forward<Args>(args)...));
```

Comment:
When we use this solution, we don't need `#include <cstdarg>`, so we can remove it